File: java/resistor-color/src/main/java/ResistorColor.java
import java.util.NoSuchElementException;

class ResistorColor {
    private final static String[] COLORS = new String[] {
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white",
    };

    int colorCode(String color) {
        for (int i = 0; i < COLORS.length; i++) {
            if (COLORS[i].equals(color)) {
                return i;
            }
        }
        throw new NoSuchElementException();
    }

    String[] colors() {
        return COLORS;
    }
}

File: c/resistor-color/src/resistor_color.c
#include "resistor_color.h"

static const resistor_band_t COLORS[] = {
    BLACK,
    BROWN,
    RED,
    ORANGE,
    YELLOW,
    GREEN,
    BLUE,
    VIOLET,
    GREY,
    WHITE
};

const resistor_band_t* colors() {
    return COLORS;
}

int color_code(resistor_band_t resistor_band) {
    return resistor_band;
}
